src/roi/Roi.ts

```diff
@@ -91,4 +145,513 @@ export class Roi {
+      for (let y = 0; y < this.height; y++) {
+        for (let x = 0; x < this.width; x++) {
+          let target =
+            (y + this.origin.row) * this.map.width + x + this.origin.column;
+          if (this.map.data[target] === this.id) {
+            points.push([x, y]);
+          }
+        }
+      }
+      this.computed.points = points;
+    }
+    return this.computed.points;
+  }
+  get boxIDs() {
+    if (!this.computed.boxIDs) {
+      this.computed.boxIDs = getBoxIDs(this);
+    }
+    return this.computed.boxIDs;
+  }
+
+  /**
+   * Returns the diameter of a circle of equal projection area
+   */
+  get eqpc() {
+    if (!this.computed.eqpc) {
+      this.computed.eqpc = 2 * Math.sqrt(this.surface / Math.PI);
+    }
+    return this.computed.eqpc;
+  }
+
+  get external() {
+    if (!this.computed.external) {
+      this.computed.external = getExternal(this);
+    }
+    return this.computed.external;
+  }
+
+  get holesInfo() {
+    if (!this.computed.holesInfo) {
+      this.computed.holesInfo = getHolesInfo(this);
+    }
+    return this.computed.holesInfo;
+  }
+
+  getExternalIDs(): void {
+    // take all the borders and remove the internal one ...
+    let borders = this.borderIDs;
+    let lengths = this.borderLengths;
+
+    this.computed.externalLengths = [];
+    this.computed.externalIDs = [];
+
+    let internals = this.internalIDs;
+
+    for (let i = 0; i < borders.length; i++) {
+      if (!internals.includes(borders[i])) {
+        this.computed.externalIDs.push(borders[i]);
+        this.computed.externalLengths.push(lengths[i]);
+      }
+    }
+  }
+
+  get borderIDs() {
+    if (this.computed.borderIDs) {
+      return this.computed.borderIDs;
+    }
+    this._computeBorderIDs();
+    return this.computed.borderIDs || [];
+  }
+
+  get borderLengths(): number[] {
+    if (this.computed.borderLengths) {
+      return this.computed.borderLengths;
+    }
+    this._computeBorderIDs();
+    return this.computed.borderLengths || [];
+  }
+
+  get box() {
+    // points of the Roi that touch the rectangular shape
+    if (!this.computed.box) {
+      this.computed.box = getBox(this);
+    }
+    return this.computed.box;
+  }
+  /**
+   * Getter that calculates fill ratio of the ROI
+   */
+  get fillRatio() {
+    return this.surface / (this.surface + this.holesInfo.surface);
+  }
+  /**
+   * Getter that calculates sphericity of the ROI
+   */
+  get sphericity() {
+    return (2 * Math.sqrt(this.surface * Math.PI)) / this.perimeter;
+  }
+
+  /**
+   * Getter that calculates solidity of the ROI
+   */
+  get solidity() {
+    return this.surface / getConvexHull(this.getMask()).surface;
+  }
+
+  get roundness() {
+    /*Slide 24 https://static.horiba.com/fileadmin/Horiba/Products/Scientific/Particle_Characterization/Webinars/Slides/TE011.pdf */
+    return (
+      (4 * this.surface) /
+      (Math.PI * this.feretDiameters.maxDiameter.length ** 2)
+    );
+  }
+
+  get feretDiameters(): {
```

Comment:
It does not only contain Diameters but also the aspectRatio. I would rename it to `feret`. Also the type should not be redefined. It exists and is returned by `getFeret` (type = `Feret`)